### ISSUE TITLE
Possible error using t_lod and t_lod_abs

### ISSUE THREAD

**Issue by jameshay218** [thimotei/legacy_ct_modelling#1]
Hi Tim, Sam and Joel,

This code is brilliant, thanks for sharing. It's proving very useful!

I am getting a bit mixed up about the way that the `t_lod` parameter is being used in the stan code vs. the post-fitting simulation code. I wonder if it's a minor mistake or if I am misinterpreting something.

My understanding is that `t_lod` is the time from peak Ct value to hitting `c_0`, the true baseline Ct value.

On [L144](https://github.com/thimotei/legacy_ct_modelling/blob/b0094e154f69208f5e1fd76f742a7c272e354c95/stan/ct_trajectory_model.stan#L144) of `ct_trajectory_model_stan`, the variable `t_lod_abs` is created, which is the time from infection to hitting the true baseline Ct (`c_0`) at the end of infection.

Then, on [L16/21](https://github.com/thimotei/legacy_ct_modelling/blob/b0094e154f69208f5e1fd76f742a7c272e354c95/stan/functions/piecewise_ct.stan#L16) of the function `piecewise_ct`, `t_lod_abs` is used to calculate the clearance rate. Shouldn't this be:

`lgrad = (clod-cs)/tlod` rather than `lgrad = (clod-cs)/t_lod_abs`?

In the simulation code of [`piecewise_ct`](https://github.com/thimotei/legacy_ct_modelling/blob/b0094e154f69208f5e1fd76f742a7c272e354c95/R/simulate.R#L24), the parameter `tlod` is instead treated as it's original meaning -- the time from peak to `c_0`. Other than on  [L27](https://github.com/thimotei/legacy_ct_modelling/blob/b0094e154f69208f5e1fd76f742a7c272e354c95/R/simulate.R#L27), which I think should be `t > tlod + te + tp + ts` rather than `t > tlod`.

TLDR; the stan code is calculating the time from peak to LOD as `tlod + tp + ts`, whereas in the simulation code it is being calculated as just `tlod`. Not sure how this is working out because all your figures and fits look good.

Cheers,
James

**Comment by jameshay218** [comment]
Related is the interpretation of the "time until LOD" parameter estimates in the manuscript figure 3 -- I think this is not the actual timing parameter being used in the model (what's being shown is actually `t_lod - t_p`, as `t_p` is added to `t_lod` in all calculations of the time to LOD. It is also time until hitting the "true" baseline level `c_0`, which is different to the assay LOD of 40. This is an important distinction, as looking at the parameters alone implies that the infection would be detectable for longer than it really is (and partially explains why these estimates are much longer than ours).

EDIT: this applies to the priors too -- this would explain partly why the posteriors are so shifted from the priors; the prior parameters have a different interpretation to the ones being fitted.

**Comment by seabbs** [comment]
Hi @jameshay218, 

Thanks for this detailed exploration and interest more generally. I think you are correct in all points here unfortunately. 

I think the mistake is in the model code where `tlod` should be used (i.e., the relative parameter) for `piecewise_ct`. As you. say it is used "correctly" in the simulation code. This is a bit surprising to me as it should have been pretty clear from fitting against simulated data so its annoying this slipped through.

I think fitting to real data you would just not see this as you wouldn't be able to spot the mismatch? 

> o time until hitting the "true" baseline level

Agree.

> -- this would explain partly why the posteriors are so shifted from the priors

Agree.

I am currently teaching in SA but will be able to look at this in more detail the week after next. 

Thanks again for the deep dive! 

Sam

**Comment by thimotei** [comment]
Hi James,

Wow, thanks so much for this deep dive. It seems you are correct! Its nothing more than a bug in the Stan code I believe. Given all of the flexibility build into the model, it didn't affect how well Stan thought it was fitting too much. However, it does change the results of the resulting time until LOD is reached parameters by a few days. The rest of the fitted parameters seem to be very similar/the same once its fit. Very glad you found this at this stage, as we're currently revising the results! It used to give one or two divergent transitions, and now it fits without any, so that is also a nice bonus!

I think Sam is right in the sense that all we need to do is fix the bug in the Stan code itself. The rest of the code seems like it should be fine (once that is fixed). If you agree, now that I've fixed the Stan code, I'll close this issue

Thanks again, this sort of deep dive is always so helpful for complicated models

**Comment by jameshay218** [comment]
Sounds good, thanks for fixing it.
